system-helper/flatpak-system-helper.c

```diff
@@ -1151,6 +1276,416 @@ handle_run_triggers (FlatpakSystemHelper   *object,
+  g_autoptr(OngoingPull) pull = NULL;
+  g_autoptr(GSubprocessLauncher) launcher = NULL;
+  g_autoptr(GSubprocess) revokefs_backend = NULL;
+  int sockets[2];
+
+  pull = g_slice_new0 (OngoingPull);
+  pull->object = object;
+  pull->invocation = invocation;
+  pull->ref = g_strdup (ref);
+  pull->commit = g_strdup (commit);
+  pull->src_dir = g_strdup (src);
+  pull->cancellable = g_cancellable_new ();
+  pull->uid = uid;
+  pull->unique_name = g_strdup (g_dbus_connection_get_unique_name (connection));
+
+  pull->watch_id = g_bus_watch_name_on_connection (connection,
+                                                   pull->unique_name,
+                                                   G_BUS_NAME_WATCHER_FLAGS_NONE, NULL,
+                                                   name_vanished_cb,
+                                                   g_strdup (g_dbus_connection_get_unique_name (connection)),
+                                                   g_free);
+
+  if (socketpair (AF_UNIX, SOCK_SEQPACKET, 0, sockets) == -1)
+    {
+      glnx_throw_errno_prefix (error, "Failed to get a socketpair");
+      return NULL;
+    }
+
+  launcher = g_subprocess_launcher_new (G_SUBPROCESS_FLAGS_NONE);
+  g_subprocess_launcher_set_child_setup (launcher, revokefs_fuse_backend_child_setup, passwd, NULL);
+  g_subprocess_launcher_take_fd (launcher, sockets[0], 3);
+  fcntl (sockets[1], F_SETFD, FD_CLOEXEC);
+  pull->client_socket = sockets[1];
+
+  pull->revokefs_backend = g_subprocess_launcher_spawn (launcher,
+                                                        error,
+                                                        "revokefs-fuse",
+                                                        "--backend",
+                                                        "--socket=3", src, NULL);
+  if (pull->revokefs_backend == NULL)
+    return NULL;
+
+  return g_steal_pointer (&pull);
+}
+
+static gboolean
+check_for_repo_containing_partial_pull (GFile          *cache_dir,
+                                        const gchar    *commit,
+                                        gchar         **out_repo_location,
+                                        struct passwd  *passwd)
+{
+  g_autoptr(GFileEnumerator) enumerator = NULL;
+  GFileInfo *file_info = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *cache_dirname = g_file_get_path (cache_dir);
+  struct stat st_buf;
+  const gchar *name;
+
+  if (stat (cache_dirname, &st_buf) == 0 &&
+      /* should be owned by SYSTEM_HELPER_USER */
+      st_buf.st_uid == passwd->pw_uid &&
+      /* should not be world-writeable */
+      (st_buf.st_mode & 0022) == 0)
+    g_debug ("Cache directory %s status valid", cache_dirname);
+  else
+    {
+      g_warning ("Cache directory %s status invalid", cache_dirname);
+      return FALSE;
+    }
+
+  enumerator = g_file_enumerate_children (cache_dir,
+                                          G_FILE_ATTRIBUTE_STANDARD_NAME,
```

Comment:
I think you need G_FILE_ATTRIBUTE_STANDARD_TYPE here too.

Reply:
replaced with 
` G_FILE_ATTRIBUTE_STANDARD_NAME  ","  G_FILE_ATTRIBUTE_STANDARD_TYPE`